File: app/src/main/java/com/ps/simplepapersoccer/GameObjects/Move/PossibleMove.java
package com.ps.simplepapersoccer.GameObjects.Move;

import com.ps.simplepapersoccer.GameObjects.Game.Node;

/**
 * Created by Admin on 2017-04-24.
 */

public class PossibleMove
{
    public PossibleMove(Node oldNode, Node newNode)
    {
        this.oldNode = oldNode;
        this.newNode = newNode;
    }

    @Override
    public boolean equals(Object object) {
        if (object == null) return false;
        if (object.getClass() != getClass()) return false;
        PossibleMove other = (PossibleMove) object;
        if (!oldNode.id.equals(other.oldNode.id)) return false;
        if (!newNode.id.equals(other.newNode.id)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return oldNode.id.hashCode() ^ newNode.id.hashCode();
    }

    public Node oldNode;
    public Node newNode;
}

File: app/src/main/java/com/ps/simplepapersoccer/GameObjects/Move/PossibleMove.kt
package com.ps.simplepapersoccer.GameObjects.Move

import com.ps.simplepapersoccer.GameObjects.Game.Node

/**
 * Created by Admin on 2017-04-24.
 */

class PossibleMove(var oldNode: Node, var newNode: Node) {

    override fun equals(`object`: Any?): Boolean {
        if (`object` == null) return false
        if (`object`.javaClass != javaClass) return false
        val other = `object` as PossibleMove?
        if (oldNode.id != other!!.oldNode.id) return false
        if (newNode.id != other.newNode.id) return false
        return true
    }

    override fun hashCode(): Int {
        return oldNode.id.hashCode() xor newNode.id.hashCode()
    }
}
